tests/integration/conftest.py

```diff
@@ -4,18 +4,52 @@
 import pandas as pd
 import pytest
 
+from great_expectations.compatibility.typing_extensions import override
+from great_expectations.data_context.data_context.context_factory import set_context
 from great_expectations.datasource.fluent.interfaces import Batch
 from tests.integration.test_utils.data_source_config import DataSourceTestConfig
+from tests.integration.test_utils.data_source_config.base import (
+    BatchTestSetup,
+    dict_to_tuple,
+    hash_data_frame,
+)
 
 _F = TypeVar("_F", bound=Callable)
 
 
 @dataclass(frozen=True)
-class _TestConfig:
+class TestConfig:
     data_source_config: DataSourceTestConfig
     data: pd.DataFrame
     extra_data: Mapping[str, pd.DataFrame]
 
+    @override
+    def __hash__(self) -> int:
+        return hash(
+            (
+                self.__class__,
+                self.data_source_config,
+                hash_data_frame(self.data),
+                dict_to_tuple(
+                    {k: hash_data_frame(self.extra_data[k]) for k in sorted(self.extra_data)}
+                ),
+            )
+        )
+
+    @override
+    def __eq__(self, value: object) -> bool:
+        # We need to implement this ourselves to call `.equals` on dataframes.`
+        if not isinstance(value, TestConfig):
+            return False
+        return all(
```

Comment:
I want to push back a bit here. I considered similar approaches. I went with this because, while more tedious, it saves us from the (very unlikely) case of hash collisions. But the likelihood of that is sooo small, so I'm totally happy to change to a hash comparison, but wanted to lay out my thoughts. LMK.

Reply:
You are right, this is better.